Auth pull request

Karsich opened the pull request.

Sug4chy committed:
Merge pull request #8 from CSU-2025-1/develop

Develop

Sug4chy committed:
Merge pull request #12 from CSU-2025-1/develop

Develop

Sug4chy committed:
Merge pull request #19 from CSU-2025-1/develop

Develop

Sug4chy committed:
Merge pull request #23 from CSU-2025-1/develop

Add migration with ResumeMakingStatus

Karsich committed:
DB done

Karsich committed:
Application + mess gprc and web

Sug4chy commented on ResumeGenerator.AuthService/ResumeGenerator.AuthService.Application/DTO/Requests/LoginUserRequest.cs:
```diff
@@ -0,0 +1,6 @@
+﻿namespace ResumeGenerator.AuthService.Application.DTO.Requests;
+
+public record LoginUserRequest(
```
sealed

Sug4chy commented on ResumeGenerator.AuthService/ResumeGenerator.AuthService.Application/DTO/Requests/RegisterUserRequest.cs:
```diff
@@ -0,0 +1,6 @@
+﻿namespace ResumeGenerator.AuthService.Application.DTO.Requests;
+
+public record RegisterUserRequest(
```
sealed

Sug4chy commented on ResumeGenerator.AuthService/ResumeGenerator.AuthService.Application/DTO/Responses/LoginUserResponse.cs:
```diff
@@ -0,0 +1,6 @@
+﻿namespace ResumeGenerator.AuthService.Application.DTO.Responses;
+
+public record LoginUserResponse(
```
sealed

Sug4chy commented on ResumeGenerator.AuthService/ResumeGenerator.AuthService.Application/DTO/Responses/LoginUserResponse.cs:
```diff
@@ -0,0 +1,6 @@
+
+public record LoginUserResponse(
+    string Token,
+    DateTime ExpiresAt
```
Это поле можно убрать, как правило в сам токен зашивается Claims с временем истечения

Sug4chy commented on ResumeGenerator.AuthService/ResumeGenerator.AuthService.Application/DTO/Responses/RegisterUserResponse.cs:
```diff
@@ -0,0 +1,6 @@
+﻿namespace ResumeGenerator.AuthService.Application.DTO.Responses;
+
+public record RegisterUserResponse(
```
sealed

Sug4chy commented on ResumeGenerator.AuthService/ResumeGenerator.AuthService.Application/DTO/Responses/RegisterUserResponse.cs:
```diff
@@ -0,0 +1,6 @@
+﻿namespace ResumeGenerator.AuthService.Application.DTO.Responses;
+
+public record RegisterUserResponse(
+    Guid UserId,
```
Тоже не нужно

Sug4chy commented on ResumeGenerator.AuthService/ResumeGenerator.AuthService.Application/Exceptions/InvalidCredentialsException.cs:
```diff
@@ -0,0 +1,7 @@
+﻿namespace ResumeGenerator.AuthService.Application.Exceptions;
+
+public class InvalidCredentialsException : Exception
```
sealed

Sug4chy commented on ResumeGenerator.AuthService/ResumeGenerator.AuthService.Application/Exceptions/UserAlreadyExistsException.cs:
```diff
@@ -0,0 +1,7 @@
+﻿namespace ResumeGenerator.AuthService.Application.Exceptions;
+
+public class UserAlreadyExistsException : Exception
```
sealed

Sug4chy commented on ResumeGenerator.AuthService/ResumeGenerator.AuthService.Application/Exceptions/UserNotActiveException.cs:
```diff
@@ -0,0 +1,7 @@
+﻿namespace ResumeGenerator.AuthService.Application.Exceptions;
+
+public class UserNotActiveException : Exception
```
sealed

Sug4chy commented on ResumeGenerator.AuthService/ResumeGenerator.AuthService.Application/Exceptions/UserNotFoundException.cs:
```diff
@@ -0,0 +1,7 @@
+﻿namespace ResumeGenerator.AuthService.Application.Exceptions;
+
+public class UserNotFoundException : Exception
```
sealed

Sug4chy commented on ResumeGenerator.AuthService/ResumeGenerator.AuthService.Application/ResumeGenerator.AuthService.Application.csproj:
```diff
@@ -0,0 +1,36 @@
+		<TargetFramework>net9.0</TargetFramework>
+		<Nullable>enable</Nullable>
+		<ImplicitUsings>enable</ImplicitUsings>
+		<RootNamespace>ResumeGenerator.AuthService.Web</RootNamespace>
```
ResumeGenerator.AuthService.Application

Sug4chy commented on ResumeGenerator.AuthService/ResumeGenerator.AuthService.Application/ResumeGenerator.AuthService.Application.csproj:
```diff
@@ -0,0 +1,36 @@
+
+	<ItemGroup>
+		<!-- Ссылки на проекты -->
+		<ProjectReference Include="..\ResumeGenerator.AuthService.Application\ResumeGenerator.AuthService.Application.csproj" />
```
Почему

Sug4chy commented on ResumeGenerator.AuthService/ResumeGenerator.AuthService.Application/ResumeGenerator.AuthService.Application.csproj:
```diff
@@ -0,0 +1,36 @@
+		<ProjectReference Include="..\ResumeGenerator.AuthService.Data\ResumeGenerator.AuthService.Data.csproj" />
+	</ItemGroup>
+
+	<ItemGroup>
```
Вот эту группу можно убрать

Sug4chy commented on ResumeGenerator.AuthService/ResumeGenerator.AuthService.Application/Services/AuthService.cs:
```diff
@@ -0,0 +1,99 @@
+
+namespace ResumeGenerator.AuthService.Application.Services;
+
+public class AuthService : IAuthService
```
sealed

Sug4chy commented on ResumeGenerator.AuthService/ResumeGenerator.AuthService.Application/Services/AuthService.cs:
```diff
@@ -0,0 +1,99 @@
+    public async Task<RegisterUserResponse> RegisterUserAsync(RegisterUserRequest request)
+    {
+        if (await _context.Users.AnyAsync(u => u.Username == request.Username))
+            throw new UserAlreadyExistsException(request.Username);
```
Кудрявые скобки для единства стиля кода

Sug4chy commented on ResumeGenerator.AuthService/ResumeGenerator.AuthService.Application/Services/AuthService.cs:
```diff
@@ -0,0 +1,99 @@
+            ExpiresAt = DateTime.UtcNow.AddDays(1)
+        };
+
+        await _context.Users.AddAsync(user);
```
Add вместо AddAsync

Sug4chy commented on ResumeGenerator.AuthService/ResumeGenerator.AuthService.Application/Services/AuthService.cs:
```diff
@@ -0,0 +1,99 @@
+        _botLinkGenerator = botLinkGenerator;
+    }
+
+    public async Task<RegisterUserResponse> RegisterUserAsync(RegisterUserRequest request)
```
Последним параметром CancelationToken ct = default

Sug4chy commented on ResumeGenerator.AuthService/ResumeGenerator.AuthService.Application/Services/AuthService.cs:
```diff
@@ -0,0 +1,99 @@
+        var user = await _context.Users
+            .FirstOrDefaultAsync(u => u.Username == request.Username);
+
+        if (user == null)
```
Кудрявые скобки тоже

Sug4chy commented on ResumeGenerator.AuthService/ResumeGenerator.AuthService.Application/Services/AuthService.cs:
```diff
@@ -0,0 +1,99 @@
+        var token = _tokenGenerator.GenerateToken(user);
+        var expiresAt = DateTime.UtcNow.AddDays(7);
+
+        var authToken = new AuthToken
```
Это в базу не сохраняем

Sug4chy commented on ResumeGenerator.AuthService/ResumeGenerator.AuthService.Application/Services/BCryptPasswordHasher.cs:
```diff
@@ -0,0 +1,14 @@
+﻿namespace ResumeGenerator.AuthService.Application.Services;
+
+public class BCryptPasswordHasher : IPasswordHasher
```
sealed

Sug4chy commented on ResumeGenerator.AuthService/ResumeGenerator.AuthService.Application/Services/IActivationCodeGenerator.cs:
```diff
@@ -0,0 +1,6 @@
+﻿namespace ResumeGenerator.AuthService.Application.Services;
+
+public interface IActivationCodeGenerator
```
Убрать

Sug4chy commented on ResumeGenerator.AuthService/ResumeGenerator.AuthService.Application/Services/IAuthService.cs:
```diff
@@ -0,0 +1,10 @@
+public interface IAuthService
+{
+    Task<RegisterUserResponse> RegisterUserAsync(RegisterUserRequest request);
+    Task<LoginUserResponse> LoginUserAsync(LoginUserRequest request);
```
В асинхронных методах всегда последним аргументом CancellationToken ct = default

Sug4chy commented on ResumeGenerator.AuthService/ResumeGenerator.AuthService.Application/Services/IBotLinkGenerator.cs:
```diff
@@ -0,0 +1,6 @@
+
+public interface IBotLinkGenerator
+{
+    string GenerateLink(string code);
```
code заменить на GuidUserId можно

Sug4chy commented on ResumeGenerator.AuthService/ResumeGenerator.AuthService.Application/Services/ITokenGenerator.cs:
```diff
@@ -0,0 +1,11 @@
+public interface ITokenGenerator
+{
+    string GenerateToken(User user);
+    ClaimsPrincipal ValidateToken(string token);
```
Переименовать в ParseToken (либо вообще удалить)

Sug4chy commented on ResumeGenerator.AuthService/ResumeGenerator.AuthService.Application/Services/JwtTokenGenerator.cs:
```diff
@@ -0,0 +1,59 @@
+
+namespace ResumeGenerator.AuthService.Application.Services;
+
+public class JwtTokenGenerator : ITokenGenerator
```
sealed

Sug4chy commented on ResumeGenerator.AuthService/ResumeGenerator.AuthService.Application/Services/JwtTokenGenerator.cs:
```diff
@@ -0,0 +1,59 @@
+
+public class JwtTokenGenerator : ITokenGenerator
+{
+    private readonly IConfiguration _configuration;
```
Целиком конфигурацию не прокидываем, лучше отдельным кусками, в этом случае чисто секцию Jwt
https://metanit.com/sharp/aspnet6/6.8.php

Sug4chy commented on ResumeGenerator.AuthService/ResumeGenerator.AuthService.Application/Services/JwtTokenGenerator.cs:
```diff
@@ -0,0 +1,59 @@
+        {
+            new Claim(ClaimTypes.NameIdentifier, user.Id.ToString()),
+            new Claim(ClaimTypes.Name, user.Username),
+            new Claim("chat_id", user.ChatId?.ToString() ?? string.Empty)
```
Убрать

Sug4chy commented on ResumeGenerator.AuthService/ResumeGenerator.AuthService.Application/Services/JwtTokenGenerator.cs:
```diff
@@ -0,0 +1,59 @@
+        var securityKey = new SymmetricSecurityKey(Encoding.UTF8.GetBytes(_configuration["Jwt:Key"]));
+        var credentials = new SigningCredentials(securityKey, SecurityAlgorithms.HmacSha256);
+
+        var claims = new[]
```
Добавить Сlaim с ExpiresAt

Sug4chy commented on ResumeGenerator.AuthService/ResumeGenerator.AuthService.Application/Services/TelegramBotLinkGenerator.cs:
```diff
@@ -0,0 +1,19 @@
+
+public class TelegramBotLinkGenerator : IBotLinkGenerator
+{
+    private readonly IConfiguration _configuration;
```
Прокидываем секцию, а не всю конфигурацию

Sug4chy commented on ResumeGenerator.AuthService/ResumeGenerator.AuthService.Application/Services/TelegramBotLinkGenerator.cs:
```diff
@@ -0,0 +1,19 @@
+
+namespace ResumeGenerator.AuthService.Application.Services;
+
+public class TelegramBotLinkGenerator : IBotLinkGenerator
```
sealed

Sug4chy commented on ResumeGenerator.AuthService/ResumeGenerator.AuthService.Application/Validators/LoginUserRequestValidator.cs:
```diff
@@ -0,0 +1,13 @@
+
+namespace ResumeGenerator.AuthService.Application.Validators;
+
+public class LoginUserRequestValidator : AbstractValidator<LoginUserRequest>
```
sealed

Sug4chy commented on ResumeGenerator.AuthService/ResumeGenerator.AuthService.Application/Validators/LoginUserRequestValidator.cs:
```diff
@@ -0,0 +1,13 @@
+    public LoginUserRequestValidator()
+    {
+        RuleFor(x => x.Username).NotEmpty();
+        RuleFor(x => x.Password).NotEmpty();
```
Можно добавить сразу для пароля правила типа длины не меньше N символов, обязательное наличие цифр, заглавных букв и тд

Sug4chy commented on ResumeGenerator.AuthService/ResumeGenerator.AuthService.Application/Validators/RegisterUserRequestValidator.cs:
```diff
@@ -0,0 +1,20 @@
+
+namespace ResumeGenerator.AuthService.Application.Validators;
+
+public class RegisterUserRequestValidator : AbstractValidator<RegisterUserRequest>
```
sealed

Sug4chy commented on ResumeGenerator.AuthService/ResumeGenerator.AuthService.Data/Context/AppDbContext.cs:
```diff
@@ -0,0 +1,18 @@
+
+public sealed class AppDbContext(DbContextOptions<AppDbContext> options) : DbContext(options)
+{
+
```
Пустую строку убрать

Sug4chy commented on ResumeGenerator.AuthService/ResumeGenerator.AuthService.Data/Entities/ActivationCode.cs:
```diff
@@ -0,0 +1,15 @@
+﻿using System.ComponentModel.DataAnnotations;
+
+namespace ResumeGenerator.AuthService.Data.Entities
+{
```
В file-scoped namespace

Sug4chy commented on ResumeGenerator.AuthService/ResumeGenerator.AuthService.Data/Entities/AuthToken.cs:
```diff
@@ -0,0 +1,14 @@
+﻿using System.ComponentModel.DataAnnotations;
+
+namespace ResumeGenerator.AuthService.Data.Entities
```
В file-scoped namespace

Sug4chy commented on ResumeGenerator.AuthService/ResumeGenerator.AuthService.Data/Entities/User.cs:
```diff
@@ -0,0 +1,18 @@
+﻿using System.ComponentModel.DataAnnotations;
+
+namespace ResumeGenerator.AuthService.Data.Entities
```
В file-scoped namespace

Sug4chy commented on ResumeGenerator.AuthService/ResumeGenerator.AuthService.Data/Entities/User.cs:
```diff
@@ -0,0 +1,18 @@
+        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
+        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
+
+        public List<AuthToken> AuthTokens { get; set; } = new();
```
Лучше вместо листа писать более общий ICollection, и вместо new() ставить []

Sug4chy commented on ResumeGenerator.AuthService/ResumeGenerator.AuthService.Data/EntityConfigurations/UserEntityConfiguration.cs:
```diff
@@ -0,0 +1,23 @@
+        builder.HasKey(h => h.Id);
+
+        builder.Property(h => h.Id).HasColumnName("id");
+        builder.Property(h => h.Username).HasColumnName("username");
```
Можно было бы ещё индекс уникальности навесить на username

Sug4chy commented on ResumeGenerator.AuthService/ResumeGenerator.AuthService.Data/ResumeGenerator.AuthService.Data.csproj:
```diff
@@ -0,0 +1,29 @@
+		<PackageReference Include="Npgsql.EntityFrameworkCore.PostgreSQL" Version="9.0.2" />
+	</ItemGroup>
+
+	<ItemGroup>
```
Можно убрать

Sug4chy commented on ResumeGenerator.AuthService/ResumeGenerator.AuthService.Data/ResumeGenerator.AuthService.Data.csproj:
```diff
@@ -0,0 +1,29 @@
+	  <Folder Include="Migrations\" />
+	</ItemGroup>
+
+	<ItemGroup>
```
Можно убрать

Sug4chy commented on ResumeGenerator.AuthService/ResumeGenerator.AuthService.Grpc/ResumeGenerator.AuthService.Grpc.csproj:
```diff
@@ -0,0 +1,16 @@
+
+	<ItemGroup>
+		<PackageReference Include="Grpc.AspNetCore" Version="2.62.0" />
+		<PackageReference Include="Microsoft.EntityFrameworkCore" Version="9.0.2" />
```
Вот эту зависимость можно убрать

Sug4chy commented on ResumeGenerator.AuthService/ResumeGenerator.AuthService.Grpc/ResumeGenerator.AuthService.Grpc.csproj:
```diff
@@ -0,0 +1,16 @@
+﻿<Project Sdk="Microsoft.NET.Sdk.Web">
```
Sdk сменить на Microsoft.NET.Sdk, чтобы это была просто библиотека классов

Sug4chy commented on ResumeGenerator.AuthService/ResumeGenerator.AuthService.Grpc/ResumeGenerator.AuthService.Grpc.csproj:
```diff
@@ -0,0 +1,16 @@
+
+	<ItemGroup>
+		<Protobuf Include="Protos\auth.proto" GrpcServices="Server" />
+		<ProjectReference Include="..\ResumeGenerator.AuthService.Data\ResumeGenerator.AuthService.Data.csproj" />
```
Будто бы лучше зависимость на Application сделать

Sug4chy commented on ResumeGenerator.AuthService/ResumeGenerator.AuthService.Web/Initializers/DatabaseMigrationInitializer.cs:
```diff
@@ -0,0 +1,24 @@
+
+namespace ResumeGenerator.AuthService.Web.Initializers;
+
+public static class DatabaseMigrationInitializer
```
Советую посмотреть в сторону Extensions.Hosting.AsyncInitialization библиотеки, чтобы свои велосипеды не изобретать

Sug4chy commented on ResumeGenerator.AuthService/ResumeGenerator.AuthService.Web/Startup.cs:
Не нужен

Sug4chy commented on ResumeGenerator.AuthService/ResumeGenerator.AuthService.Web/Program.cs:
```diff
@@ -1,41 +1,42 @@
 }
 
 app.UseHttpsRedirection();
+app.UseAuthorization();
```
Перед UseAuthorization ещё должен быть вызов метода UseAuthentication

Karsich committed:
Fixed application mess

Добавил sealed, ощутил мощь прокидывания секций а не всей конфигурации, стер activation codes с лица земли

Karsich committed:
Fixed data

Karsich committed:
grpc + small changes

Sug4chy commented on ResumeGenerator.AuthService/ResumeGenerator.AuthService.Application/Configurations/JwtOptions.cs:
```diff
@@ -0,0 +1,11 @@
+{
+    public const string SectionName = "Jwt";
+
+    public string Key { get; set; } = string.Empty;
```
Вот тут set замени на init, чтобы случайно даже не было возможности менять

Sug4chy commented on ResumeGenerator.AuthService/ResumeGenerator.AuthService.Application/Services/AuthService.cs:
```diff
@@ -0,0 +1,114 @@
+            UpdatedAt = DateTime.UtcNow
+        };
+
+        await _context.Users.AddAsync(user, ct);
```
Просто Add вместо AddAsync всегда в EF Core

Sug4chy commented on ResumeGenerator.AuthService/ResumeGenerator.AuthService.Application/Services/AuthService.cs:
```diff
@@ -0,0 +1,114 @@
+
+        if (user == null)
+        {
+            return new ActivationResult(false, "Пользователь не найден");
```
Тут разве не исключение должно у тебя лететь, если пользователь не найден?

Sug4chy commented on ResumeGenerator.AuthService/ResumeGenerator.AuthService.Application/Services/AuthService.cs:
```diff
@@ -0,0 +1,114 @@
+        var principal = _tokenGenerator.ValidateToken(token);
+        var userId = principal.Claims.First(x => x.Type == ClaimTypes.NameIdentifier).Value;
+
+        var user = await _context.Users.FindAsync(new object[] { Guid.Parse(userId) }, ct);
```
FindAsync немного непредсказуемая вещь, потому что он сначала лезет в кэш контекста, и потом уже шлёт запрос в БД. А что там в кэше - хрен его разбери, так что лучше поменяй на обычный запрос в БД, как ты раньше делал. С FirstOrDefaultAsync и исключением, если не нашёл

Sug4chy commented on ResumeGenerator.AuthService/ResumeGenerator.AuthService.Application/Services/AuthService.cs:
```diff
@@ -0,0 +1,114 @@
+        var user = await _context.Users.FindAsync(new object[] { Guid.Parse(userId) }, ct);
+
+        if (user == null)
+            throw new InvalidOperationException("Пользователь не найден");
```
Кудрявые скобки

Sug4chy commented on ResumeGenerator.AuthService/ResumeGenerator.AuthService.Application/Services/TelegramBotLinkGenerator.cs:
```diff
@@ -0,0 +1,20 @@
+
+    public string GenerateLink(string guidUserId)
+    {
+        return _options.DeepLinkTemplate
```
Не понимаю, зачем тут этот DeepLinkTemplate, если честно. В конфиге юзернейма бота достаточно, его прокидывай да подставляй

Sug4chy commented on ResumeGenerator.AuthService/ResumeGenerator.AuthService.Data/Entities/AuthToken.cs:
Это мы вроде договорились удалить

Sug4chy commented on ResumeGenerator.AuthService/ResumeGenerator.AuthService.Grpc/AuthGrpcService.cs:
```diff
@@ -0,0 +1,41 @@
+        var result = await _authService.ActivateUserAsync(request.ActivationCode, context.CancellationToken);
+        return new ActivateUserResponse
+        {
+            Success = result.Success,
```
Вот эти Success не надо городить. У gRPC тоже есть статус-коды свои, поэтому и из сервиса также возвращай нормально ошибки исключениями. И сделать Interceptor, типа Мидлвару, чтобы он их ловил и переводил в статус-коды

Sug4chy commented on ResumeGenerator.AuthService/ResumeGenerator.AuthService.Web/Startup.cs:
У тебя в Program.cs всё добавляется, этот файл скопированный удали просто

Karsich committed:
mvp auth

Sug4chy commented on ResumeGenerator.AuthService/ResumeGenerator.AuthService.Application/DTO/Responses/RegisterUserResponse.cs:
```diff
@@ -0,0 +1,5 @@
+﻿namespace ResumeGenerator.AuthService.Application.DTO.Responses;
+
+public sealed record RegisterUserResponse(
+    Guid UserId
```
Ссылка на бота, ID возвращать ни к чему

Sug4chy commented on ResumeGenerator.AuthService/ResumeGenerator.AuthService.Application/Services/IBotLinkGenerator.cs:
```diff
@@ -0,0 +1,6 @@
+
+public interface IBotLinkGenerator
+{
+    string GenerateLink(string guidUserId);
```
Принмать Guid в аргументах

Karsich committed:
Fixing review issues

Karsich committed:
DONE

Sug4chy commented on ResumeGenerator.AuthService/ResumeGenerator.AuthService.Application/Configurations/JwtOptions.cs:
```diff
@@ -0,0 +1,11 @@
+    public const string SectionName = "Jwt";
+
+    public string Key { get; init; } = string.Empty;
+    public string Issuer { get; set; } = string.Empty;
```
init я имел в виду везде поставить)

Sug4chy commented on ResumeGenerator.AuthService/ResumeGenerator.AuthService.Application/Configurations/TelegramBotOptions.cs:
```diff
@@ -0,0 +1,8 @@
+{
+    public const string SectionName = "TelegramBot";
+
+    public string Username { get; set; } = string.Empty;
```
И тут init тоже

Sug4chy commented on ResumeGenerator.AuthService/ResumeGenerator.AuthService.Application/Services/AuthService.cs:
```diff
@@ -0,0 +1,115 @@
+        }
+            
+        user.IsActive = true;
+        user.UpdatedAt = DateTime.UtcNow;
```
Вот такие вещи, кстати, можно вынести в Interceptor, чтобы при сохранении в базу они сами проставлялись. Если захочешь добавить, то скину пример, но в целом и так норм

Sug4chy commented on ResumeGenerator.AuthService/ResumeGenerator.AuthService.Application/Services/IAuthService.cs:
```diff
@@ -0,0 +1,13 @@
+    Task<LoginUserResponse> LoginUserAsync(LoginUserRequest request, CancellationToken ct = default);
+    Task ActivateUserAsync(string activationCode, CancellationToken ct = default);
+    Task<UserDto> GetUserByTokenAsync(string token, CancellationToken ct = default);
+    public record UserDto(Guid Id, string Username, bool IsActive);
```
Лучше отдельно вынеси

Sug4chy commented on ResumeGenerator.AuthService/ResumeGenerator.AuthService.Web/Controllers/AuthController.cs:
```diff
@@ -0,0 +1,90 @@
+    public async Task<IActionResult> Register([FromBody] RegisterUserRequest request)
+    {
+
+        if (!ModelState.IsValid)
```
От этого if нет смысла, потому что ModelState у тебя нигде не проверяется. Тут надо или ручками делать if-ы и добавлять ошибки в ModelState, либо валидировать через data-аннотации. А вообще, лучше просто заведи себе FluentValidation библиотеку, тут из DI вытаскивай валидатор и если не гуд, то тогда уже 400-ку

Sug4chy commented on ResumeGenerator.AuthService/ResumeGenerator.AuthService.Web/Controllers/AuthController.cs:
```diff
@@ -0,0 +1,90 @@
+    [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+    public async Task<IActionResult> Login([FromBody] LoginUserRequest request)
+    {
+        if (!ModelState.IsValid)
```
Тут такая же история про валидацию

Karsich committed:
Done?

Sug4chy commented on ResumeGenerator.AuthService/ResumeGenerator.AuthService.Web/Program.cs:
```diff
@@ -1,41 +1,55 @@
+builder.Services.Configure<TelegramBotOptions>(builder.Configuration.GetSection(TelegramBotOptions.SectionName));
+
+builder.Services.AddControllers();
+builder.Services.AddValidatorsFromAssembly(typeof(IAuthService).Assembly); //������������ ��� ���������� �� application
```
Подожди, а сами классы валидаторов-то где...? Или я пропустил

Karsich committed:
Cool Validation
